docs/resources/rule_set.md

```diff
@@ -0,0 +1,139 @@
+---
+subcategory: "Security"
+---
+
+# databricks_rule_set Resource
+
+This resource allows you to manage access rules on Databricks account level resources. For convenience we allow accessing this resource through the Databricks account and workspace.
+
+To configure rule sets through the Databricks account, the provider must be configured with `host = "https://accounts.cloud.databricks.com"` on AWS deployments or `host = "https://accounts.azuredatabricks.net"` and authenticate using [AAD tokens](https://registry.terraform.io/providers/databricks/databricks/latest/docs#special-configurations-for-azure) on Azure deployments
+
+## Example usage
+
+Rule set management through AWS Databricks workspace:
+
+```hcl
+locals {
+  account_id = "00000000-0000-0000-0000-000000000000"
+}
+
+// account level group
+data "databricks_group" "ds" {
+  display_name = "Data Science"
+}
+
+resource "databricks_service_principal" "automation_sp" {
+  display_name = "SP_FOR_AUTOMATION"
+}
+
+resource "databricks_rule_set" "automation_sp_rule_set" {
```

Comment:
Name identifies a rule set resource, we also set ID to be equal to name. How does terraform pass the grant_rules to us?

Reply:
My question was about following scenario - if I define two resources with the same name and different principals - what will happen?

```hcl
resource "databricks_rule_set" "automation_sp_rule_set1" {
  name = "accounts/${local.account_id}/servicePrincipals/${databricks_service_principal.automation_sp.application_id}/ruleSets/default"
  grant_rules {
    principals = ["groups/${data.databricks_group.ds1.display_name}"]
    role       = "roles/servicePrincipal.user"
  }
}

resource "databricks_rule_set" "automation_sp_rule_set2" {
  name = "accounts/${local.account_id}/servicePrincipals/${databricks_service_principal.automation_sp.application_id}/ruleSets/default"
  grant_rules {
    principals = ["groups/${data.databricks_group.ds2.display_name}"]
    role       = "roles/servicePrincipal.user"
  }
}
```

from the code I see that it should attempt to merge both definitions, but I want to be sure that I understand how it works, especially if you remove only one resource - what will happen with another?

Also, what will happen if I have intersecting principals, etc.?

Reply:
why is having resource with same name but different principal required?

Reply:
> if I define two resources with the same name and different principals - what will happen?

Ah yeah, this can be problematic. The behavior will be unpredictable because terraform possibly thinks the rule set resources can be created / updated in parallel. The warning suggestion sounds good to me. Let me add it. 

> Also, what will happen if I have intersecting principals, etc.?

As mentioned above, there will be unpredictable behavior.